File: examples/dagger-project-server-with-server/src/main/java/io/github/jojoti/examples/dagger/di/AppComponent.java
package io.github.jojoti.examples.dagger.di;

import dagger.Component;
import io.github.jojoti.examples.dagger.dao.DaoModule;
import io.github.jojoti.examples.dagger.router.RouterApi;
import io.github.jojoti.examples.dagger.service.ServiceModule;

/**
 * @author JoJo Wang
 * @link github.com/jojoti
 * @date 2020/5/18
 */
@Component(modules = {AppModule.class, ServiceModule.class, DaoModule.class})
@AppScope
public interface AppComponent {

    void injectApiHandler(RouterApi routerApi);

}

File: util/share-id-v2/src/main/kotlin/io/github/jojoti/util/shareid/AppComponent.kt
package io.github.jojoti.util.shareid

import dagger.Component
import javax.inject.Singleton

/**
 * @author JoJo Wang
 * @link github.com/jojoti
 */
@Singleton
@Component(modules = [AppModule::class])
interface AppComponent {

    fun hashShareId(): HashShareId

}